packages/socket.io-promise/index.d.ts

```diff
@@ -0,0 +1,32 @@
+export interface IoPromiseAction {
+  type: string;
+}
+
+export interface IoPromiseSocket {
+  emit(event: string, ...args: any[]): any;
```

Comment:
I did this mostly just to cut down on another outside dependency, especially since we really only use a very small part of the `socket.io` API for this middleware. Since we never return a socket and only use it as an input, we don't really lose anything by using a smaller interface. I guess it means people could pass things that don't implement the whole `socket.io` API and just have an `emit` method, but that seems like a weird edge case.